### PR TITLE
Solution by Kate

### PR DESCRIPTION
**Pull request by ekaterinakur**
[Preview] (https://ekaterinakur.github.io/js_numberdrome/)

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+class Numberdrome {
+  constructor() {
+    this.numbersList = [];
+  }
+
+  addNumber(n) {
+    if (typeof n !== 'number' || !isFinite(n)) {
+      return;
+    }
+
```
Comment (gestap) [inline src/main.js]: remove empty line

File: src/main.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  max() {
+    try {
+      if (!this.numbersList.length) {
+        throw new Error("Incorrect data: can`t find maximum of empty array");
+      };
+
+      return Math.max(...this.numbersList);
+    } catch(error) {
+      return error.message;
```
Comment (gestap) [inline src/main.js]: you need to `throw an error` instead of `return message`

File: src/main.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  min() {
+    try {
+      if (!this.numbersList.length) {
+        throw new Error("Incorrect data: can`t find minimum of empty array");
+      };
+
+      return Math.min(...this.numbersList);
+    } catch(error) {
+      return error.message;
```
Comment (gestap) [inline src/main.js]: you need to `throw an error` instead of `return message`

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  removeNumber(n) {
+    this.numbersList.splice(this.numbersList.indexOf(n), 1);
+  }
+
+  sum() {
+    let sum = 0;
+    if (this.numbersList.length) {
+      this.numbersList.forEach(num => sum += num);
```
Comment (gestap) [inline src/main.js]: better to use `reduce` for this task

File: src/main.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    return sum;
+  }
+
+  product() {
+    let product = 1;
+    if (this.numbersList.length) {
+      this.numbersList.forEach(num => product *= num);
```
Comment (gestap) [inline src/main.js]: better to use `reduce` for this task

**Review by ost-ptk — CHANGES_REQUESTED** [review]

File: src/main.js (hunk 1 above)
Comment (ekaterinakur) [reply src/main.js]: Done

File: src/main.js (hunk 2 above)
Comment (ekaterinakur) [reply src/main.js]: Done

File: src/main.js (hunk 3 above)
Comment (ekaterinakur) [reply src/main.js]: Done

File: src/main.js (hunk 4 above)
Comment (ekaterinakur) [reply src/main.js]: Done

File: src/main.js (hunk 5 above)
Comment (ekaterinakur) [reply src/main.js]: Done

**Review by ost-ptk — APPROVED** [review]

**Pull request closed without merging**